src/perception/semantic_segmentation/semantic_segmentation/segmentation_node.py

```diff
@@ -0,0 +1,142 @@
+import numpy as np
+import cv2
+import os
+from mmseg.apis import MMSegInferencer
+import torch
+import rclpy
+from rclpy.node import Node
+from rclpy.qos import QoSProfile, QoSReliabilityPolicy, QoSHistoryPolicy
+from sensor_msgs.msg import Image, CompressedImage
+from std_msgs.msg import String
+from cv_bridge import CvBridge
+from rich.logging import RichHandler
+import logging
+from ament_index_python.packages import get_package_share_directory
+# Configure the root logger
+logging.basicConfig(level=logging.INFO, handlers=[RichHandler(level=logging.WARNING)])
+
+
+# ALGO_VERSION = os.getenv("MODEL_NAME")
+
+# if not ALGO_VERSION:
+#     ALGO_VERSION = 'nvidia/segformer-b2-finetuned-ade-512-512'
+
+package_name = 'semantic_segmentation'
+package_share_directory = get_package_share_directory(package_name)
```

Comment:
this should happen in the launch file, same place as where the yaml config is loaded